app/src/main/java/com/karumi/SuperHeroesApplication.kt

```diff
@@ -33,8 +40,13 @@ class SuperHeroesApplication : Application(), KodeinAware {
     private fun appDependencies(): Module {
         return Module(allowSilentOverride = true) {
             bind<SuperHeroRepository>() with singleton {
-                SuperHeroRepository()
+                SuperHeroRepository(listOf(MemorySuperHeroDataSource(instance()),
```

Comment:
Shouldn't be the memory data source the only singleton here? (instead of the repository itself)

Reply:
fixed